common/src/main/java/dev/architectury/extensions/injected/InjectedBlockEntityTypeExtension.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * This file is part of architectury.
+ * Copyright (C) 2020, 2021, 2022 architectury
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package dev.architectury.extensions.injected;
+
+import dev.architectury.registry.registries.Registries;
+import net.minecraft.core.Registry;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+
+public interface InjectedBlockEntityTypeExtension extends InjectedRegistryEntryExtension {
+    @Override
+    default ResourceLocation arch$getRegistryName() {
+        return Registries.getId((BlockEntityType<?>) this, Registry.BLOCK_ENTITY_TYPE_REGISTRY);
+    }
```

Comment:
In my opinion this should be using the builtInRegistryHolder, given that that is why we wanted to add this extension in the first place

(My originally proposed structure was going to be a `WithBuiltInHolder<T>` interface which just relies on the already existing builtInRegistryHolder method, and that would then be used for methods such as getRegistryKey, which would return a `ResourceKey<T>`, and getRegistryName, which would return the ResourceLocation)